src/meta/meta_bulk_load_service.cpp

```diff
@@ -706,12 +706,22 @@ void bulk_load_service::handle_app_pausing(const bulk_load_response &response,
 // ThreadPool: THREAD_POOL_META_STATE
 void bulk_load_service::try_rollback_to_downloading(const std::string &app_name, const gpid &pid)
 {
-    zauto_read_lock l(_lock);
+    zauto_write_lock l(_lock);
+
     const auto app_status = get_app_bulk_load_status_unlocked(pid.get_app_id());
     if (app_status == bulk_load_status::BLS_DOWNLOADING ||
         app_status == bulk_load_status::BLS_DOWNLOADED ||
         app_status == bulk_load_status::BLS_INGESTING ||
         app_status == bulk_load_status::BLS_SUCCEED) {
```

Comment:
I think App in the ingesting state could not be rolled back. How can you make rocksdb ingestion rolled back?

Reply:
The app in ingesting state should also rollback, Ingestion requires all replicas have sst files. If a replica server crash, partitions on this server will be unavailable, those partitions should rollback to downloading to ensure all replicas have downloaded files succeed.
Rocksdb ingestion don't support rollback.

Reply:
Ok. An INGESTING app being rolled-back to DOWNLOADING is reasonsable. But why does a SUCCEED app also need rollback? If I don't misunderstand, SUCCEED means all partitions having the ingested data. The data won't lost even when all the ReplicaServers are down. This can be assured because ingestion is a 2PC.

Reply:
The app whose state is succeed are cleaning its downloaded files, and app `is_bulk_loading` is still equal to true. In current implementation, succeed will also rollback to downloading if meet recoverable errors. It may be optimized in next version.

Reply:
You can add a comment here later. I think this case is trivial but needs notice.

Reply:
You can see bulk load process graph in `meta_bulk_load_service.h`, it shows possible status convert.